webapp/src/ts/services/enketo.service.ts

```diff
@@ -629,16 +629,38 @@ interface XmlFormContext {
   instanceData: null|string|Record<string, any>; // String for report forms, Record<> for contact forms.
   titleKey?: string;
   isFormInModal?: boolean;
-  contactSummary: Record<string, any>;
+  contactSummary?: Record<string, any>;
 }
 
-export interface EnketoFormContext {
+export class EnketoFormContext {
   selector: string;
   formDoc: Record<string, any>;
-  instanceData: null|string|Record<string, any>; // String for report forms, Record<> for contact forms.
+  type: 'contact'|'report'|'task'|'training-card';
+  editing: boolean;
+  instanceData: null|string|Record<string, any>;
   editedListener: () => void;
   valuechangeListener: () => void;
   titleKey?: string;
   isFormInModal?: boolean;
   userContact?: Record<string, any>;
+  contactSummary? :Record<string, any>;
+
+  constructor(selector?, type?, formDoc?, instanceData?) {
```

Comment:
Minor, but since this class is getting pretty broad usage, it would be nice to be more specific about the types of the params here and which are _required_.  (e.g. I am pretty sure we should always be getting the first three....)

Reply:
Thinking about this more, I wonder if we should just remove instanceData from the constructor. It can get set in the same way as the other optional params....

Reply:
Just need a builder pattern here to go with the class....  😂

Reply:
I added `instanceData` after getting frustrated by e2e tests.